lib/msf/ui/console/command_dispatcher/db.rb

```diff
@@ -1786,36 +1799,27 @@ def cmd_db_connect(*args)
       end
     end
 
-    if yaml_file
-      if (yaml_file and not ::File.exist? ::File.expand_path(yaml_file))
-        print_error("File not found")
-        return
-      end
-      file = yaml_file || ::File.join(Msf::Config.get_config_root, "database.yml")
-      file = ::File.expand_path(file)
-      if (::File.exist? file)
-        db = YAML.load(::File.read(file))['production']
-        framework.db.connect(db)
-        print_line('Connected to the database specified in the YAML file.')
-        return
-      end
+    result = Msf::DbConnector.db_connect(framework, opts)
+    if result[:error]
+      print_error result[:error]
+      return
     end
 
-    meth = "db_connect_#{new_conn_type}"
-    if(self.respond_to?(meth, true))
-      self.send(meth, opts)
-    else
-      print_error("This database driver #{new_conn_type} is not currently supported")
+    if result[:result]
+      print_status result[:result]
     end
-
     if framework.db.active
+      name = opts[:name]
       if !name || name.empty?
         if found_name
           name = found_name
+        elsif result[:data_service_name]
+          name = result[:data_service_name]
         else
           name = Rex::Text.rand_text_alphanumeric(8)
         end
       end
```

Comment:
The name mechanism in this command dispatcher seems slightly broken/inconsistent, particularly with the mismatch of not storing local db creds within `~/.msf4/config` itself. I've fixed some bugs/edge cases on the way, but overall I think it needs more work - but not as part of this pull request